Hunting Queries/TeamsLogs/NewExternalOrg.yaml

```diff
@@ -0,0 +1,33 @@
+id: 6fce5baf-bfc2-4c56-a6b7-9c4733fc5a45
+name: External user from a new organisation added
+description: |
+  'This query identifies external users added to Teams where the user's domain is not one 
+  previously seen in Teams data.
+  This query requires you to have Teams data collected from the O365 Management Activity API
+  and the Teams parser function enabled.'
+tactics:
+  - Persistance
+relevantTechniques:
+  - T1136
+query: |
+
+  // If you have more than 30 days worth of Teams data change this value
+  let data_date = 30d;
+  // If you want to look at users further back than the last day change this value
+  let lookback_data = 1d;
+  let known_orgs = (
+  TeamsData 
+  | where TimeGenerated > ago(data_date)
+  | where Operation == "MemberAdded" or Operation == "TeamsSessionStarted"
+  // Extract the correct UPN and parse our external organization domain
+  | extend UPN = iif(Operation == "MemberAdded", tostring(parse_json(Members)[0].UPN), UserId)
+  | extend Organization = tostring(split(split(UPN, "_")[1], "#")[0])
```

Comment:
We also have the format for UPN different than what you parse above for Organization - "user_contoso.com#EXT#@M365xXXXXX.onmicrosoft.com" that we may consider handling.  This format also comes thru as - user@M365xXXXXX.onmicrosoft.com.  Should we be handling this?

Reply:
So we are only handling and parsing external users were, which all have the "user_contoso.com#EXT#@M365xXXXXX.onmicrosoft.com" format.
 user@M365xXXXXX.onmicrosoft.com are native tenant users so we are not interested in including them in this query.